### DISCUSSION TITLE
Input Body for Indexer Report API

### DISCUSSION THREAD

**Discussion by imaskm** [quay/clair#1255] [Q&A] [answered]
Hi,
while calling indexer_report api with post request, we need to pass something like below in body:
`const inputBody = '{
  "hash": "sha256:fc84b5febd328eccaa913807716887b3eb5ed08bc22cc6933a9ebf82766725e3",
  "layers": [
    {
      "hash": "sha256:fc84b5febd328eccaa913807716887b3eb5ed08bc22cc6933a9ebf82766725e3",
      "uri": "https://storage.example.com/blob/2f077db56abccc19f16f140f629ae98e904b4b7d563957a7fc319bd11b82ba36\n",
      "headers": {
        "property1": [
          "string"
        ],
        "property2": [
          "string"
        ]
      }
    }
  ]
}';`

In this data, how can one get all the URIs for respective hash both in private and public docker registry?

Thanks in advance!!

**Comment by hdonnay** [comment, marked as answer]
Every registry should support the OCI [distribution](https://github.com/opencontainers/distribution-spec/blob/main/spec.md) and [image](https://github.com/opencontainers/image-spec/blob/master/spec.md) specifications; those can be used if you're not integrating specifically with a registry. Authz for private layers will depend on specifics, but signed URLs and header tokens should work as expected.

The `clairctl` tool has a `manifest` subcommand which can create manifests from public container repositories, and should understand the `.docker/config.json` file for private repositories.
